ddtrace/contrib/trace_utils.py

```diff
@@ -479,10 +480,15 @@ def set_http_meta(
         # We always collect the IP if appsec is enabled to report it on potential vulnerabilities.
         # https://datadoghq.atlassian.net/wiki/spaces/APS/pages/2118779066/Client+IP+addresses+resolution
         if config._appsec_enabled or config.retrieve_client_ip:
-            ip = _get_request_header_client_ip(request_headers, peer_ip, headers_are_case_sensitive)
-            if ip:
-                span.set_tag_str(http.CLIENT_IP, ip)
-                span.set_tag_str("network.client.ip", ip)
+            # Retrieve the IP if it was calculated on AppSecProcessor.on_span_start
+            request_ip = _context.get_item("http.request.remote_ip", span=span)
+
+            if not request_ip:
+                # Not calculated: framework does not support IP blocking or testing env
+                request_ip = _get_request_header_client_ip(request_headers, peer_ip, headers_are_case_sensitive)
+
+            span.set_tag_str(http.CLIENT_IP, request_ip)
+            span.set_tag_str("network.client.ip", request_ip)
```

Comment:
do we need both of these tags?

seems redundant to set the same value twice under two different tags. (I see it was here before, so maybe not worth getting into here)

Reply:
It seems we do since they're used on different specs but I need to speak with some people to see if we're to late to unify stuff. Out of the PR context anyway.